Source/DafnyDriver/Commands/DafnyCli.cs

```diff
@@ -480,6 +482,14 @@ class WritersConsole : IConsole {
       //  "Usage:\ndafny format [--check] [--print] <file/folder> <file/folder>...\nYou can use '.' for the current directory");
       return ExitValue.PREPROCESSING_ERROR;
     }
+    
+    // Add standard library .doo files after explicitly provided source files,
+    // only because if they are added first, one might be used as the program name,
+    // which is not handled well.
+    if (options.Get(CommonOptionBag.AllowStdLibs)) {
+      options.CliRootSourceUris.Add(StandardLibrariesDooUri);
```

Comment:
If you want this to work for `dafny server`, you will need to update `DafnyProject.GetRootSourceUris` so it returns `StandardLibrariesDooUri`.

Alternatively, I can do the refactoring to reuse more of the server code for the CLI and then this'll come for free.

Reply:
I'd be very happy with that - I'm fine with the examples and tests in DafnyStandardLibraries not resolving in the IDE initially, since that doesn't block adding new libraries, but would love to fast-follow with fixing that. I suspect that refactoring will fix the awkwardness too since the DafnyFile will be added from the root URI automatically?